connectivity/tests/errors.go

```diff
@@ -126,7 +145,52 @@ func computeExpectedDropReasons(defaultReasons, inputReasons []string) string {
 	return filter
 }
 
-func (n *noErrorsInLogs) checkErrorsInLogs(logs string, t *check.Test) {
+func (n *noErrorsInLogs) allCiliumPods(ctx context.Context, ct *check.ConnectivityTest) (map[podID]podInfo, error) {
+	output := make(map[podID]podInfo)
+
+	// List all Cilium-related pods
+	for _, client := range ct.Clients() {
+		pods, err := client.ListPods(ctx, ct.Params().CiliumNamespace, metav1.ListOptions{LabelSelector: ct.Params().CiliumPodSelector})
+		if err != nil {
+			return nil, err
+		}
+
+		cluster := client.ClusterName()
+		for _, pod := range pods.Items {
+			pod := pod
+			output[podID{Cluster: cluster, Namespace: pod.Namespace, Name: pod.Name}] = podInfo{
+				client: client, containers: n.podContainers(&pod),
+			}
+		}
+	}
+
+	// Additionally add Cilium agent pods, if not already included in the previous
+	// list. This prevents missing them in case the "all cilium pods" selector does
+	// not match any pod (mainly in v1.12, as the app.kubernetes.io/part-of=cilium
+	// label was not yet present at that time).
+	for _, pod := range ct.CiliumPods() {
+		id := podID{Cluster: pod.K8sClient.ClusterName(), Namespace: pod.Namespace(), Name: pod.NameWithoutNamespace()}
+		if _, ok := output[id]; !ok {
+			output[id] = podInfo{client: pod.K8sClient, containers: n.podContainers(pod.Pod)}
+		}
+	}
+
+	return output, nil
+}
+
+func (n *noErrorsInLogs) podContainers(pod *corev1.Pod) (containers []string) {
```

Comment:
nit: No prealloc of cos slice. 
Also wondering if these are all agent pods or operator pods likely the container list should always be the same?

Reply:
I'd personally prefer keep this generic and simple, avoiding to hardcode the knowledge about the number of containers associated with any pods (as it may change between versions). I'd also not overoptimize at this point, given that this logic is executed only once, and the collection of logs is orders of magnitude more expensive in any case.